commit db6d1da22e65327ae29dcf9bc50986b2e8f3f0bd
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Thu Oct 1 01:05:23 2026 +0200

    hurd: Fix xfailing tst-pututxline and tst-arc4random-fork

diff --git a/sysdeps/mach/hurd/Makefile b/sysdeps/mach/hurd/Makefile
index dcc4d0fb4a..44779fe019 100644
--- a/sysdeps/mach/hurd/Makefile
+++ b/sysdeps/mach/hurd/Makefile
@@ -253,9 +253,6 @@ test-xfail-tst-mallocfork2-malloc-check = yes
 test-xfail-tst-mallocfork2-malloc-hugetlb1 = yes
 test-xfail-tst-mallocfork2-malloc-hugetlb2 = yes
 test-xfail-tst-mallocfork2-malloc-largetcache = yes
-test-xfail-tst-pututxline-cache = yes
-test-xfail-tst-pututxline-lockfail = yes
-test-xfail-tst-arc4random-fork = yes
 endif
 ifeq ($(subdir),htl)
 test-xfail-tst-mutex4 = yes
@@ -269,12 +266,17 @@ test-xfail-tst-cond23 = yes
 test-xfail-tst-rwlock4 = yes
 test-xfail-tst-rwlock12 = yes
 test-xfail-tst-barrier2 = yes
-test-xfail-tst-pututxline-cache = yes
-test-xfail-tst-pututxline-lockfail = yes
 test-xfail-tst-flock2 = yes
 test-xfail-tst-signal1 = yes
 test-xfail-tst-signal2 = yes
 endif
+ifeq ($(subdir),login)
+test-xfail-tst-pututxline-cache = yes
+test-xfail-tst-pututxline-lockfail = yes
+endif
+ifeq ($(subdir),stdlib)
+test-xfail-tst-arc4random-fork = yes
+endif
 
 ifeq ($(subdir),htl)
 # For bug 25522
